src/app/server/Server.cpp

```diff
@@ -327,9 +327,6 @@ void Server::FactoryReset(intptr_t arg)
     // Emit Shutdown event, as shutdown will come after factory reset.
     DispatchShutDownEvent(0);
 
-    // Flush all dispatched events.
-    chip::app::InteractionModelEngine::GetInstance()->GetReportingEngine().ScheduleUrgentEventDeliverySync();
-
     chip::DeviceLayer::ConfigurationMgr().InitiateFactoryReset();
```

Comment:
As far as I can tell, factory reset (at least in the implementations I spot-checked) does not do an ordely shutdown of the Matter stack (and in particular does not call reporting::Engine::Shutdown).  Was this codepath tested?  On what platforms?

Reply:
Yes, that's why we called `ScheduleUrgentEventDeliverySync` explicitly.

Reply:
Yep, just revert this accidental removal., thanks!